smartsim/entity/application.py

```diff
@@ -0,0 +1,288 @@
+        """
+        self._files = copy.deepcopy(value)
+
+    @property
+    def file_parameters(self) -> t.Mapping[str, str]:
+        """Return file parameters.
+
+        :returns: application file parameters
+        """
+        return self._file_parameters
+
+    @file_parameters.setter
+    def file_parameters(self, value: t.Mapping[str, str]) -> None:
+        """Set the file parameters.
+
+        :param value: file parameters
+        """
+        self._file_parameters = copy.deepcopy(value)
+
+    @property
+    def incoming_entities(self) -> t.List[SmartSimEntity]:
+        """Return incoming entities.
+
+        :returns: incoming entities
+        """
+        return self._incoming_entities
+
+    @incoming_entities.setter
+    def incoming_entities(self, value: t.List[SmartSimEntity]) -> None:
+        """Set the incoming entities.
+
+        :param value: incoming entities
+        """
+        self._incoming_entities = copy.deepcopy(value)
+
+    @property
+    def key_prefixing_enabled(self) -> bool:
+        """Return whether key prefixing is enabled for the application.
+
+        :param value: key prefixing enabled
+        """
+        return self._key_prefixing_enabled
+
+    @key_prefixing_enabled.setter
+    def key_prefixing_enabled(self, value: bool) -> None:
+        """Set whether key prefixing is enabled for the application.
+
+        :param value: key prefixing enabled
+        """
+        self.key_prefixing_enabled = copy.deepcopy(value)
+
+    def add_exe_args(self, args: t.Union[str, t.List[str], None]) -> None:
+        """Add executable arguments to executable
+
+        :param args: executable arguments
+        """
+        args = self._build_exe_args(args)
+        self._exe_args.extend(args)
+
+    def attach_generator_files(
+        self,
+        to_copy: t.Optional[t.List[str]] = None,
+        to_symlink: t.Optional[t.List[str]] = None,
+        to_configure: t.Optional[t.List[str]] = None,
+    ) -> None:
+        """Attach files to an entity for generation
+
+        Attach files needed for the entity that, upon generation,
+        will be located in the path of the entity.  Invoking this method
+        after files have already been attached will overwrite
+        the previous list of entity files.
+
+        During generation, files "to_copy" are copied into
+        the path of the entity, and files "to_symlink" are
+        symlinked into the path of the entity.
+
+        Files "to_configure" are text based application input files where
+        parameters for the application are set. Note that only applications
+        support the "to_configure" field. These files must have
+        fields tagged that correspond to the values the user
+        would like to change. The tag is settable but defaults
+        to a semicolon e.g. THERMO = ;10;
+
+        :param to_copy: files to copy
+        :param to_symlink: files to symlink
+        :param to_configure: input files with tagged parameters
```

Comment:
Do we need a ``:raises:`` doc string for the ``ValueError``?

Reply:
added